### PR TITLE
Output check result and details as JSON if requested

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: code/EnvironmentChecker.php [hunk 1]
```diff
@@ -88,11 +88,18 @@ function canAccess($member = null, $permission = "ADMIN") {
 	function index() {
 		$response = new SS_HTTPResponse;
 		$result = EnvironmentCheckSuite::inst($this->checkSuiteName)->run();
-		
+
+		// output the result as JSON if requested
+		if($this->getRequest()->getVar('format') == 'json') {
```
Comment (stojg) [inline code/EnvironmentChecker.php]: would it be better to check for the standard http header `Content-Type: application/json;` ?

A lot of REST interfaces works like that.

Comment (stojg) [inline code/EnvironmentChecker.php]: This is how I sort of faked it in the deploynaut api:


		if($this->getRequest()->getExtension() == 'json') {
			return true;
		}
		if(strpos($this->getRequest()->getHeader('Accept'), 'application/json') !== false) {
			return true;
		}

Comment (halkyon) [inline code/EnvironmentChecker.php]: Thanks, I think that is better than what I had :) I've updated the PR.

File: code/EnvironmentChecker.php [hunk 2]
```diff
@@ -88,11 +88,21 @@ function canAccess($member = null, $permission = "ADMIN") {
...
-		
+
+		// output the result as JSON if requested
+		if(
+			$this->getRequest()->getExtension() == 'json'
+			|| strpos($this->getRequest()->getHeader('Accept'), 'application/json') !== false
+		) {
+			$response->setBody($result->toJSON());
+			$response->addHeader('Content-Type', 'application/json');
+			return $response;
```
Comment (mateusz) [inline code/EnvironmentChecker.php]: Hmm this will bypass the emailing functionality below... is this what we want?

Cough `defer sendEmail()` oh wait...

Comment (halkyon) [inline code/EnvironmentChecker.php]: Quite right sir. I've fixed that up. `defer` in PHP would be nice for sure :)

**Pull request merged**